app/src/main/java/com/quran/labs/androidquran/BookmarksWidget.kt

```diff
@@ -0,0 +1,65 @@
+package com.quran.labs.androidquran
+
+import android.app.PendingIntent
+import android.appwidget.AppWidgetManager
+import android.appwidget.AppWidgetProvider
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.widget.RemoteViews
+import com.quran.labs.androidquran.ui.PagerActivity
+import com.quran.labs.androidquran.ui.QuranActivity
+import com.quran.labs.androidquran.widgets.WidgetService
+
+class BookmarksWidget : AppWidgetProvider() {
+
+  override fun onUpdate(context: Context, appWidgetManager: AppWidgetManager, appWidgetIds: IntArray) {
+    for (appWidgetId in appWidgetIds) {
+      val serviceIntent = Intent(context, WidgetService::class.java).apply {
+        putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, appWidgetId)
+        data = Uri.parse(toUri(Intent.URI_INTENT_SCHEME))
+      }
+
+      val widget = RemoteViews(context.packageName, R.layout.bookmarks_widget)
+
+      var intent = Intent(context, QuranDataActivity::class.java)
+      var pendingIntent = PendingIntent.getActivity(context, 0, intent, 0)
```

Comment:
i'd vote for val over var whenever possible since it makes things cleaner generally - though as you said, in this particular case, it's not bad - but if you want to, i wouldn't mind :)